src/Symfony/Component/HttpKernel/RebootableInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel;
+
+/**
+ * Rebotable extends the Kernel to allow it to reboot using a temporary cache directory.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+interface RebootableInterface
```

Comment:
Should this interface be merged into `HttpKernel` for 4.0? It seems a lot of stuff will rely on the `reboot()` method with this PR, maybe there's no sense to keep it in a separate interface and still require kernels to implement it?

(note that my comment would apply specifically for 4.0, not 3.x)